SMLHelper/Options/MenuAttribute.cs

```diff
@@ -0,0 +1,142 @@
+    ///     public KeyCode KeybindKey;
+    /// 
+    ///     [Slider("My slider", 0, 50, DefaultValue = 25, Format = "{0:F2}"), OnChange(nameof(MyGenericValueChangedEvent))]
+    ///     public float SliderValue;
+    /// 
+    ///     [Toggle("My checkbox"), OnChange(nameof(MyCheckboxToggleEvent)), OnChange(nameof(MyGenericValueChangedEvent))]
+    ///     public bool ToggleValue;
+    /// 
+    ///     [Button("My button")]
+    ///     public void MyButtonClickEvent(ButtonClickedEventArgs e)
+    ///     {
+    ///         Logger.Log(Logger.Level.Info, "Button was clicked!");
+    ///         Logger.Log(Logger.Level.Info, $"{e.Id}");
+    ///     }
+    /// 
+    ///     public void MyCheckboxToggleEvent(ToggleChangedEventArgs e)
+    ///     {
+    ///         Logger.Log(Logger.Level.Info, "Checkbox value was changed!");
+    ///         Logger.Log(Logger.Level.Info, $"{e.Value}");
+    ///     }
+    /// 
+    ///     private void MyGenericValueChangedEvent(IModOptionEventArgs e)
+    ///     {
+    ///         Logger.Log(Logger.Level.Info, "Generic value changed!");
+    ///         Logger.Log(Logger.Level.Info, $"{e.Id}: {e.GetType()}");
+    /// 
+    ///         switch (e)
+    ///         {
+    ///             case KeybindChangedEventArgs keybindChangedEventArgs:
+    ///                 Logger.Log(Logger.Level.Info, keybindChangedEventArgs.KeyName);
+    ///                 break;
+    ///             case ChoiceChangedEventArgs choiceChangedEventArgs:
+    ///                 Logger.Log(Logger.Level.Info, choiceChangedEventArgs.Value);
+    ///                 break;
+    ///             case SliderChangedEventArgs sliderChangedEventArgs:
+    ///                 Logger.Log(Logger.Level.Info, sliderChangedEventArgs.Value.ToString());
+    ///                 break;
+    ///             case ToggleChangedEventArgs toggleChangedEventArgs:
+    ///                 Logger.Log(Logger.Level.Info, toggleChangedEventArgs.Value.ToString());
+    ///                 break;
+    ///         }
+    ///      }
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="ChoiceAttribute"/>
+    /// <seealso cref="OnChangeAttribute"/>
+    /// <seealso cref="KeybindAttribute"/>
+    /// <seealso cref="SliderAttribute"/>
+    /// <seealso cref="ToggleAttribute"/>
+    /// <seealso cref="ButtonAttribute"/>
+    /// <seealso cref="ModOptions"/>
+    /// <seealso cref="ConfigFile"/>
+    /// <seealso cref="Logger"/>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public sealed class MenuAttribute : Attribute
+    {
+        /// <summary>
+        /// Specifies after which events the config file should be saved to disk automatically.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration has a <see cref="FlagsAttribute"/> that allows a bitwise combination of its member values.
+        /// </remarks>
+        [Flags]
+        public enum SaveEvents : short
```

Comment:
As we discussed in discord. would like a LoadEvents added as well.

Reply:
7cf2892 :)

Reply:
For your use-case, I recommend setting `LoadOn = MenuAttribute.LoadEvents.MenuRegistered | MenuAttribute.LoadEvents.MenuOpened` as this will load the config both on game start and whenever the options menu is opened.